.github/workflows/build-sphinx.yml

```diff
@@ -3,89 +3,109 @@ on:
   push:
     branches:
       - master
+  pull_request:
 
 jobs:
   build-and-deploy:
     name: Build and Deploy Docs
+
     runs-on: ubuntu-20.04
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    env:
+      python-ver: '3.9'
+
     steps:
       - name: Cancel Previous Runs
-        uses: styfle/cancel-workflow-action@0.6.0
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
           access_token: ${{ github.token }}
+
       - name: Install Intel repository
         run: |
           wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
           sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
           rm GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
           sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
           sudo apt-get update
+
       - name: Update libstdc++-dev
         run: |
           sudo apt remove -y gcc-7 g++-7 gcc-8 g++-8 gcc-10 g++-10
           sudo apt remove -y libstdc++-10-dev
           sudo apt autoremove
           sudo apt install --reinstall -y gcc-9 g++-9 libstdc++-9-dev
+
       - name: Install Intel OneAPI
         run: |
           sudo apt-get install intel-oneapi-mkl                \
                                intel-oneapi-mkl-devel          \
                                intel-oneapi-dpcpp-cpp-compiler
+
       # https://github.com/marketplace/actions/checkout
       - name: Install nvidia-cuda support drivers
         run: |
           sudo add-apt-repository ppa:graphics-drivers/ppa
           sudo apt-get update
           sudo apt-get install -y libnvidia-gl-450
           sudo apt-get install -y nvidia-cuda-toolkit clinfo
+
       - name: Checkout repo
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+
       # https://github.com/marketplace/actions/setup-miniconda
       - name: Setup miniconda
         uses: conda-incubator/setup-miniconda@v2
         with:
-          activate-environment: dpnp
-          python-version: 3.8
-          channels: intel,conda-forge
-          auto-activate-base: false
-      - name: Conda info
-        shell: bash -l {0}
-        run: |
-          conda info
-          conda list
+          auto-update-conda: true
+          python-version: ${{ env.python-ver }}
+          miniconda-version: 'latest'
+          activate-environment: 'docs'
+          channels: intel, conda-forge
+
       - name: Install sphinx dependencies
-        shell: bash -l {0}
-        run: |
-          conda install sphinx sphinx_rtd_theme
+        run: conda install sphinx sphinx_rtd_theme
+
       - name: Install dpnp dependencies
-        shell: bash -l {0}
         run: |
           conda install dpctl mkl-devel-dpcpp tbb-devel dpcpp_linux-64 cmake=3.19 cython pytest \
```

Comment:
Is there a reason to pin cmake to 3.19 ? I think newer cmake may do just as well

Reply:
I agree, we can remove the pinning.